porcelain/wallet.go

```diff
@@ -27,3 +34,27 @@ func WalletBalance(ctx context.Context, plumbing walletPlumbing, addr address.Ad
 
 	return act.Balance, nil
 }
+
+// DefaultWalletAddress returns a default wallet address from the config.
+// If none is set it picks the first address in the wallet and
+// sets it as the default in the config.
+func DefaultWalletAddress(plumbing walletPlumbing) (address.Address, error) {
+	ret, err := plumbing.ConfigGet("wallet.defaultAddress")
+	addr := ret.(address.Address)
+	if err != nil || addr != (address.Address{}) {
+		return addr, err
+	}
+
+	// No default is set; pick the 0th and make it the default.
+	if len(plumbing.WalletAddresses()) > 0 {
+		addr := plumbing.WalletAddresses()[0]
+		err := plumbing.ConfigSet("wallet.defaultAddress", addr.String())
+		if err != nil {
+			return address.Address{}, err
+		}
+
+		return addr, nil
+	}
+
+	return address.Address{}, ErrNoDefaultFromAddress
```

Comment:
this is so new you could never have know about this, but we now prefer to return address.Undef instead of the empty struct.

Reply:
I've fix it yesterday, thanks.